src/main/scala/com/bazaarvoice/sswf/model/history/HistoryFactory.scala

```diff
@@ -147,7 +160,9 @@ object HistoryFactory {
             val scheduledStep: ScheduledStep[StepEnum] = scheduledSteps(eventId)
 
             assert(scheduledStep == currentStep, s"id[$scheduledStep] != currentStep[$currentStep]")
-            Some(StepEvent[StepEnum](eventId, h.getEventId, Left(scheduledStep), StepResult.serialize(result), workflowStartTime, Some(endTime), new Duration(currentStepStart, endTime)))
+            currentStep = null
+            currentStepStart = null
+            Some(StepEvent[StepEnum](eventId, h.getEventId, Left(scheduledStep), StepResult.serialize(result), workflowStartTime, Some(endTime), new Duration(currentStepStart, endTime), invocations))
```

Comment:
right, otherwise the cumulative step duration won't be correct if task fails. I will fix this.